### ISSUE TITLE
Bitbucket Server

### ISSUE THREAD

**Issue by wirmar** [pd4d10/octohint#15]
When trying to use this on a [Bitbucket Server](https://www.atlassian.com/software/bitbucket/server), it does not work. I added `http://companyname-git:7990/*` to the options.
(Tested on github, not an installation issue)

Is Bitbucket Server supported? Could the port in the pattern be an issue?
Is there a way to tell if octohint is active on a page (possibly changing the extension icon when active)?

Chrome 63.0.3239.84
octohint 2.0.1

**Comment by pd4d10** [comment]
> Is Bitbucket Server supported?

Yes, but I only test it at https://bitbucket.org. I'm not sure if old version of Bitbucket server is supported

> Could the port in the pattern be an issue?

According to [this](https://developer.chrome.com/extensions/match_patterns), seems that `http://companyname-git/*` is OK. Port does not matter

> Is there a way to tell if octohint is active on a page (possibly changing the extension icon when active)?

It is a good idea. Will add it to future plan.

**Comment by wirmar** [comment]
> I'm not sure if old version of Bitbucket server is supported

That seems to be the Problem. I have `Atlassian Bitbucket v5.0.2` (May 2017).
It has URLs with `/browse/`->`/raw/` instead of `/src/`->`/raw/` and different css classes on the code container.
Would you be interested in an `OldBitbucketRenderer`/`BitbucketServerRenderer`?
I could try to add something if you want it, but I don't have experience in browser extensions, so any tips on setting up the dev environment would be appriciated :)

Edit: on further research it seems Bitbucket Server and Bitbucket (online) are different Products that just share the same name, so it might not be a Version issue.

**Comment by pd4d10** [comment]
@wirmar Dev environment setting up is easy, just see `scripts` field of `package.json`

In brief, for Chrome, just run `npm run dev` to start build process, output files are located at `chrome/dist`, then follow [this guide](https://developer.chrome.com/extensions/getstarted#unpacked) to load it.

I've add a `CONTRIBUTING.md` to repo, and will update it soon

Thanks! If you get stuck at something, you could ask me any time :)

**Comment by lucabezerra** [comment]
Hitching a ride on the OP's issue here :)

First of all, awesome idea, thanks for implementing it!

Secondly, on Bitbucket it seems to work only on the original files, but not those same files displayed on pull requests. Is this the intended behavior? Or is there something I need to configure in order to make it work on PRs as well?

Thanks again!

**Comment by pd4d10** [comment]
@lucabezerra 
> Is this the intended behavior?

Yes, code of pull requests are not supported currently. It is also mentioned at #9. We need to do some research for it.

Thanks!

**Comment by lucabezerra** [comment]
Oh, totally skipped that issue, went straight into the Bitbucket one, haha. Thanks for answering!
